lib/internal/modules/package_json_reader.js

```diff
@@ -73,20 +73,14 @@ function read(jsonPath, { base, specifier, isESM } = kEmptyObject) {
     let parsed;
     try {
       parsed = JSONParse(string);
-    } catch (error) {
-      if (isESM) {
-        throw new ERR_INVALID_PACKAGE_CONFIG(
-          jsonPath,
-          (base ? `"${specifier}" from ` : '') + fileURLToPath(base || specifier),
-          error.message,
-        );
-      } else {
-        // For backward compat, we modify the error returned by JSON.parse rather than creating a new one.
-        // TODO(aduh95): make it throw ERR_INVALID_PACKAGE_CONFIG in a semver-major with original error as cause
-        error.message = 'Error parsing ' + jsonPath + ': ' + error.message;
-        error.path = jsonPath;
-        throw error;
-      }
+    } catch (cause) {
+      const error = new ERR_INVALID_PACKAGE_CONFIG(
+        jsonPath,
+        isESM && (base ? `"${specifier}" from ` : '') + fileURLToPath(base || specifier),
```

Comment:
Can we get rid of isESM?

Reply:
Maybe? If I simply remove it, it doesn't work because both `base` and `specifier` are undefined, which doesn't work for `fileURLToPath`. Why do you want to get rid of it?

Reply:
After https://github.com/nodejs/node/pull/48605, this `isESM` check will be the only distinction between loaders. I'm just trying to make them consistent